presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -164,6 +158,12 @@ private void startLoadSplits()
                         startLoadSplits();
                     }
                 }
+                if (outstandingTasks.decrementAndGet() == 0) {
+                    if (fileIterators.isEmpty() && partitions.isEmpty()) {
+                        hiveSplitSource.finished();
+                        return;
```

Comment:
removed that, besides, it's better to put `partitions.isEmpty()` check after `fileIterators.isEmpty()`, it reduces the contention on `partitions` when `fileIterators` is not empty, because it's not evaluated in that case, make sense?